jsettlers.graphics/src/main/java/jsettlers/graphics/map/controls/original/panel/MainPanel.java

```diff
@@ -364,9 +364,9 @@ private void goBack() {
 
 	public void setMapViewport(MapRectangle screenArea, IGraphicsGrid grid) {
 		this.grid = grid;
-		displayCenter = new ShortPoint2D(screenArea.getLineStartX(screenArea.getLines() / 2)
-				+ screenArea.getLineLength() / 2, screenArea
-				.getLineY(screenArea.getLines() / 2));
+		short x = (short) (screenArea.getMinX() + (screenArea.getWidth() / 2));
```

Comment:
Use ints here - shorts for local variables don't save any space. ShortPoint2D has an int constructor for convenience.